example/session/src/main/java/org/apache/iotdb/DataMigrationExample.java

```diff
@@ -82,19 +96,36 @@ public static void main(String[] args)
     }
     readerPool.closeResultSet(schemaDataSet);
 
-    schemaDataSet = readerPool.executeQueryStatement("show timeseries " + path);
+    schemaDataSet = readerPool.executeQueryStatement("show devices " + path + ".**");
     schemaIter = schemaDataSet.iterator();
 
     List<Future> futureList = new ArrayList<>();
     int count = 0;
+    int seriesNumInOneTask = 0;
+    String currentDevice = "";
+    List<String> measurementsInCurrentDevice = new ArrayList<>();
+    List<TSDataType> dataTypesInCurrentDevice = new ArrayList<>();
     while (schemaIter.next()) {
+      Path currentPath = new Path(schemaIter.getString("devices"), true);
+      currentDevice = currentPath.getDevice();
+      SessionDataSetWrapper currentDivice_DataSet =
+          readerPool.executeQueryStatement("show timeseries " + currentPath + ".**");
+      DataIterator currentDivice_Iter = currentDivice_DataSet.iterator();
+      while (currentDivice_Iter.next()) {
+        Path currenttimePath = new Path(currentDivice_Iter.getString("timeseries"), true);
+        measurementsInCurrentDevice.add(currenttimePath.getMeasurement());
+        dataTypesInCurrentDevice.add(TSDataType.valueOf(currentDivice_Iter.getString("dataType")));
```

Comment:
Hi, if there is a large number of measurements, like over 100,000 measurements, in one device, is there any problem?